src/Chapter2.hs

```diff
@@ -620,7 +649,11 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [a] -> [a]
+takeEven [] = []
+takeEven (x:[]) = x : []
+takeEven (x:_:[]) = x : []
```

Comment:
This pattern is actually redundant. See, it has the same shape as the pattern on the next line. And when `xs` is empty list per the last case, function will call `takeEven` on an empty list and will return empty list as well. So you don't need to handle it separately. But it is very helpful for pattern-matching understanding to cover special and smaller cases first, so you can see how to generalize your pattern.

Reply:
I see, the thing was my implementation failed the first times and I supposed it had to do with patterns, so I went over-exhaustive and forgot to re-check 😄 Thanks!